### ISSUE TITLE
Descriptor resolution should own arguments passed to descriptors

### ISSUE THREAD

**Issue by a23bc0d5-ffdb-42d1-b9e0-05244b380ebc** [python/cpython#90719]
BPO | [46561](https://bugs.python.org/issue46561)
--- | :---
Nosy | @rhettinger, @markshannon, @tekknolagi, @sweeneyde
PRs | <li>python/cpython#30979</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2022-01-28.07:06:14.782>
labels = ['expert-C-API', '3.8', '3.9', '3.10', '3.11']
title = 'Descriptor resolution should own arguments passed to descriptors'
updated_at = <Date 2022-02-01.15:21:47.460>
user = 'https://github.com/tekknolagi'
```

bugs.python.org fields:
```python
activity = <Date 2022-02-01.15:21:47.460>
actor = 'rhettinger'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['C API']
creation = <Date 2022-01-28.07:06:14.782>
creator = 'tekknolagi'
dependencies = []
files = []
hgrepos = []
issue_num = 46561
keywords = ['patch']
message_count = 7.0
messages = ['411978', '411979', '412035', '412040', '412049', '412250', '412277']
nosy_count = 4.0
nosy_names = ['rhettinger', 'Mark.Shannon', 'tekknolagi', 'Dennis Sweeney']
pr_nums = ['30979']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue46561'
versions = ['Python 3.8', 'Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by rhettinger** [comment]
Marking this as closed.  If another core dev wants to champion it, feel free to reopen.

**Comment by serhiy-storchaka** [comment]
This was fixed in #118362.
